Yoast/Yoast-SEO-for-TYPO3
# [TASK] Introduced UrlService and LocaleService to prevent duplicate code

## Summary

This PR can be summarized in the following changelog entry:

* Duplicate code from Classes/Backend/PageLayoutHeader and Classes/Form/Element/SnippetPreview has been moved to two separate services, LocaleService and UrlService.

## Relevant technical choices:

* Constant FE_PREVIEW_TYPE has been moved to UrlService
* The check for routeEnhancers has been moved to UrlService, the error is checked with a property within

## Test instructions

This PR can be tested by following these steps:

* Checkout the changes, clear all caches (**IMPORTANT: the change within the ext_localconf produces a fatal error because of an unknown constant when you don't) and see if it all still works.

## Quality assurance
